test/e2e/istio_test.go

```diff
@@ -0,0 +1,212 @@
+		t.Errorf("failed to update labels of the namespace details: %v", err)
+	}
+
+	addToFrameworkSchemeForSmokeTests(t)
+}
+
+func (suite *IstioTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+// LIFECYCLE - Test
+
+func (suite *IstioTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *IstioTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+func TestIstioSuite(t *testing.T) {
+	suite.Run(t, new(IstioTestSuite))
+}
+
+func (suite *IstioTestSuite) TestEnvoySidecar() {
+	// First deploy a Jaeger instance
+	jaegerInstanceName := "simplest"
+	jaegerInstance := createJaegerInstanceFromFile(jaegerInstanceName, "../../examples/simplest.yaml")
+	defer undeployJaegerInstance(jaegerInstance)
+	err := WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout+(1*time.Minute))
+	require.NoError(t, err)
+
+	// Now deploy examples/business-application-injected-sidecar.yaml
+	businessAppCR := getBusinessAppCR(err)
+	defer os.Remove(businessAppCR.Name())
+	cmd := exec.Command("kubectl", "create", "--namespace", namespace, "--filename", businessAppCR.Name())
+	output, err := cmd.CombinedOutput()
+	if err != nil && !strings.Contains(string(output), "AlreadyExists") {
+		require.NoError(t, err, "Failed creating Jaeger instance with: [%s]\n", string(output))
+	}
+	const vertxDeploymentName = "myapp"
+	err = WaitForDeployment(t, fw.KubeClient, namespace, vertxDeploymentName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Failed waiting for myapp deployment")
+
+	// Add a liveliness probe to create some traces
+	vertxPort := intstr.IntOrString{IntVal: 8080}
+	livelinessHandler := &corev1.HTTPGetAction{Path: "/", Port: vertxPort, Scheme: corev1.URISchemeHTTP}
+	handler := &corev1.Handler{HTTPGet: livelinessHandler}
+	livelinessProbe := &corev1.Probe{Handler: *handler, InitialDelaySeconds: 1, FailureThreshold: 3, PeriodSeconds: 10, SuccessThreshold: 1, TimeoutSeconds: 1}
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		vertxDeployment, err := fw.KubeClient.AppsV1().Deployments(namespace).Get(context.Background(), vertxDeploymentName, metav1.GetOptions{})
+		require.NoError(t, err)
+		containers := vertxDeployment.Spec.Template.Spec.Containers
+		for index, container := range containers {
+			if container.Name == vertxDeploymentName {
+				vertxDeployment.Spec.Template.Spec.Containers[index].LivenessProbe = livelinessProbe
+				updatedVertxDeployment, err := fw.KubeClient.AppsV1().Deployments(namespace).Update(context.Background(), vertxDeployment, metav1.UpdateOptions{})
+				if err != nil {
+					log.Warnf("Error %v updating vertx app, retrying", err)
+					return false, nil
+				}
+				log.Infof("Updated deployment %v", updatedVertxDeployment.Name)
+				return true, nil
+			}
+		}
+		return false, errors.New("Vertx deployment not found")
+	})
+	require.NoError(t, err)
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
As this test case is pretty long could you also extract this to its own function?

Reply:
I have refactored this function, as well the original `wasUsingOtelAgent` one since they are very similar.